book/website/reproducible-research/code-reuse/code-reuse-packages.md

```diff
@@ -0,0 +1,164 @@
+- A package created just for fun!
+
+Packages are the fundamental units of reproducible code for programming languages and they can be useful to share publicly or even if it's a package that you use just for yourself. 
+One such package which started out as a personal library is the [HMisc](https://cran.r-project.org/web/packages/Hmisc/index.html) package (Harrell Miscellaneous) - a statistics package with a collection of useful functions maintained by Professor Frank Harrell that is now widely used amongst R users. 
+
+## What standards should be adhered to when writing a package?
+If you plan on writing a package that you wish for others to use,there are numerous considerations to make. 
+As with any other code that you want to share or just use personally, you will want to ensure your code will work reliably - and on other systems when sharing with others. 
+Some recommendations would be:
+
+- {ref}`Unit Testing <rr-testing-unittest>` and {ref}`Intergration Testing <rr-testing-types-integrationtest>` testing to ensure code within the package is robust and provides useful feedback to the end user while using it (warnings, errors).
+- {ref}`Version Control <rr-vcs>` of the codebase to track development and fix bugs.
+- {ref}`Documentation <rr-rdm-metadata>`. This might be in the form of a website or a wiki. There are even language specific packages that aim to provide project templates that will render the template into a website or other form of documentation (see [packagedown](https://pkgdown.r-lib.org/) for R, [Sphinx](https://www.sphinx-doc.org/en/master/) for Python or [Doxygen](https://www.doxygen.nl/) for C++).
+- Host your versioned codebase somewhere accessible. There are a whole range of places to host code depending on your use case. Public GitHub/GitLab repositories are extremely common especially in open source, where private repositories can be used for proprietary packages (think Matlab). Even just a local git repository might be enough for your use-case. 
+- Use Continuous Integration /  Continuous Development principles and pipelines such as GitHub Actions, Jenkins, Travis or GitLab Runners to help with a range of testing procedures.
+ - Consider submitting your package to a Central Resource such as those in the [](central-resources) above. There are usually strict requirements needed as your code is more likely to be distributed and used (more on this later on). Often GitHub repositories are used to store both stable and "daily"/"nightly" builds of a package in addition to a central resource so that users can submit bugs an contribute to the current version, while testing experimental versions as a preview. 
+ - Write a publication. This could be in the form of a a journal submission such as Numpy (in Nature!), or an abstract/paper in a conference proceeding. This allows you to peer review your package with experts in the field of intended use and might give more confidence to potential users that the package is robust.
```

Comment:
```suggestion
 - Write a publication. This could be in the form of a journal submission such as Numpy (in Nature!), or an abstract/paper in a conference proceeding. This allows your package to be peer-reviewed by experts in the field of intended use and might give more confidence to potential users that the package is robust.
```